trial-config-service/src/main/java/uk/ac/ox/ndph/mts/trial_config_service/controller/TrialConfigController.java

```diff
@@ -12,95 +14,89 @@
-    public TrialConfigController(TrialConfigService trialConfigService, WebClient webClient, GitRepo gitRepo) {
+    public TrialConfigController(TrialConfigService trialConfigService, WebClient webClient) {
         this.trialConfigService = trialConfigService;
         this.webClient = webClient;
-        this.gitRepo = gitRepo;
-    }
-
-    public TrialConfigController(TrialConfigService trialConfigService, String baseUrl) {
-        this.trialConfigService = trialConfigService;
-        this.webClient = WebClient.create(baseUrl);
     }
 
     //TODO(darrensmithson) - we are using a GET and POST for similar functionlaity to accomodate
     // early testing needs, once we are closer to final developmemnt we should ensure only 1 POST is used.
     @GetMapping("/trial")
-    public Trial saveTrialFromGit(
+    public Trial saveTrialFromConfigServer(
             @RequestParam
-                    String filename) throws InvalidConfigException, ResourceAlreadyExistsException {
-        return trialConfigService.saveTrial(createTrialFromGitRepo(filename), userId);
+                    String filename)
+        throws InvalidConfigException, ResourceAlreadyExistsException {
+        return trialConfigService.saveTrial(createTrial(filename, repoURL()), userId);
     }
 
     @PostMapping("/trial")
     public Trial saveTrialFromJson(
             @RequestBody
                     String jsonData) throws InvalidConfigException, ResourceAlreadyExistsException {
-        return trialConfigService.saveTrial(createTrialFromJsonData(jsonData), userId);
-    }
-
-    protected Trial createTrialFromGitRepo(String fileName) {
-        Trial trial;
-
-        try {
-            byte[] fileBytes = gitRepo.getTrialFile(fileName);
-            ObjectMapper objMapper = new ObjectMapper();
-            trial = objMapper.readValue(fileBytes, Trial.class);
-        } catch (IOException ioException) {
-            throw new InvalidConfigException(ioException.getMessage());
-        }
-
-        return trial;
+        return trialConfigService.saveTrial(createTrial(jsonData), userId);
     }
 
-    protected Trial createTrialFromJsonData(String trialConfig) {
+    protected Trial createTrial(String jsonData) {
         Trial trial;
 
         try {
             ObjectMapper objMapper = new ObjectMapper();
-            trial = objMapper.readValue(trialConfig, Trial.class);
+            trial = objMapper.readValue(jsonData, Trial.class);
         } catch (JsonProcessingException jpeEx) {
             throw new InvalidConfigException(jpeEx.getMessage());
         }
         return trial;
     }
 
-    protected Mono<Trial> createTrialFromURL(String trialConfig) {
-        Mono<Trial> response;
+    protected Trial createTrial(String trialConfig, String url) {
+        Trial trial;
 
         try {
-            response = webClient.get()
-                .uri(trialConfig)
+            trial = webClient.get()
+                .uri(url + File.separator + trialConfig)
```

Comment:
1. imo, revert so it won't conflict with the refactoring efforts.
2. same comment about local dev as in the practitioner service

Reply:
I don't think we should be concerned with refactoring efforts for this merge into the 341 branch - even if we decide to NOT include any code refactoring to existing Controller classes we can make that decision when we decide to merge 341 into Main

Reply:
I don't think we should be concerned with refactoring efforts for this merge into the 341 branch - even if we decide to NOT include any code refactoring to existing Controller classes we can make that decision when we decide to merge 341 into Main

Reply:
I'm ok with leaving it, if you prefer to.
however I think it will cost in a problematic merge, also I am not sure yet how will the git config repo will look like. and by having 2 clients (trial config and practitioner) we might have to refactor two instead of one microservice.
I would keep this PR to minimum. up to you

Reply:
still think refactoring should remain. I think Practitioner will be refactored soon anyway